src/ferc_xbrl_extractor/xbrl.py

```diff
@@ -84,9 +84,11 @@ def _dedupe_newer_report_wins(df: pd.DataFrame, primary_key: list[str]) -> pd.Da
     old_index = df.index.names
     return (
         df.reset_index()
+        .sort_values("report_date")
         .groupby(unique_cols)
-        .apply(lambda df: df.sort_values("report_date").ffill().iloc[-1])
+        .last()
```

Comment:
`groupby.last()` [selects the last non-null value from each column](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.DataFrameGroupBy.last.html).

Reply:
Perfect! That's the exact intent & also this conveys what's going on way better!